infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/TlsURLRewriter.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.openshift;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.URLRewriter;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+
+/** @author Ilya Buziuk */
+public class TlsURLRewriter implements URLRewriter {
+  private final boolean isTlsEnabled;
+
+  @Inject
+  TlsURLRewriter(@Named("che.infra.openshift.enable_tls") boolean isTlsEnabled) {
+    this.isTlsEnabled = isTlsEnabled;
+  }
+
+  @Override
+  public String rewriteURL(final RuntimeIdentity identity, final String name, final String url)
+      throws InfrastructureException {
+    String rewrittenURL = url;
+    if (isTlsEnabled) {
```

Comment:
Consider next refactoring:
```
if (isTlsEnabled) {
    rewrittenURL = rewriteURL(url);
}
```
```
private String rewriteURL(final String url) {
    if (isWebSocketURL(url)) {
        url.replaceFirst("^ws:", "wss:");
    } else {
        url.replaceFirst("^http:", "https:");
    }
}
```

Reply:
This might make code easier for inderstanding

Reply:
true